llvm/cmake/modules/TableGen.cmake

```diff
@@ -191,9 +191,13 @@ macro(add_tablegen target project)
 
   # FIXME: Quick fix to reflect LLVM_TABLEGEN to llvm-min-tblgen
   if("${target}" STREQUAL "llvm-min-tblgen"
-      AND NOT "${LLVM_TABLEGEN}" STREQUAL ""
-      AND NOT "${LLVM_TABLEGEN}" STREQUAL "llvm-tblgen")
-    set(${project}_TABLEGEN_DEFAULT "${LLVM_TABLEGEN}")
+      AND NOT "${LLVM_TABLEGEN}" STREQUAL "")
+    # Extract base filename from full path.
+    get_filename_component(RAW_LLVM_TABLEGEN ${LLVM_TABLEGEN} NAME_WE)
```

Comment:
I wonder `get_filename_component` would be applicable here.
Would it match if `${LLVM_TABLEGEN}` points the explicit out-of-tree `/path/to/llvm-tblgen.exe`?

`${LLVM_HOST_EXECUTABLE_SUFFIX}` may be available here, I guess.

Reply:
I'm not sure I understand the question.  For me, when `${LLVM_TABLEGEN}` is of the form:
`C:\path\to\dir-within-build-dir\bin\llvm-tblgen.exe`

the line:
`get_filename_component(RAW_LLVM_TABLEGEN ${LLVM_TABLEGEN} NAME_WE)`

sets `RAW_LLVM_TABLEGEN` to simply `llvm-tblgen`, which is what I want -- essentially, enabling the change you made in 95d4506dda79d49e55fdd0e4da7bf81487167aa1.